File: exercism/kotlin/acronym/src/main/kotlin/acronym.kt
object Acronym {

    fun generate(phrase: String): String {
        var return_string = ""
        var input_string = Regex("[^A-Za-z0-9 ]").replace(phrase, " ")
        input_string.toUpperCase().split(" ").forEach {
            if (it.length > 0) return_string += it[0]
        }
        return return_string
    }
}

File: exercism/elixir/acronym/acronym.exs
defmodule Acronym do
  @doc """
  Generate an acronym from a string.
  "This is a string" => "TIAS"
  """
  @spec abbreviate(String.t()) :: String.t()
  def abbreviate(string) do
	string |> (&(Regex.scan(~r/\b[a-z]|[A-Z]/, &1))).()
	       |> Enum.join 
	       |> String.upcase 
  end
end
